### PR TITLE
doc: Update doc versions, add release notes for 2.1.4

### PR DESCRIPTION
**Pull request by umapraseeda**
Update doc versions, known issues, add release notes for 2.1.4

### CODE REVIEW

**Review by peknis — CHANGES_REQUESTED** [review]

File: doc/nrf/releases/release-notes-2.1.4.rst [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+* Updated MPSL by removing calls to ``__WFE()`` for the nRF5340 SoC, to be in line with the Errata for the Anomaly 160.
+* Fixed a rare issue that could cause an assert when requesting timeslots with type ``MPSL_TIMESLOT_REQ_TYPE_EARLIEST``.
+
+For details, see the :ref:`MPSL changelog <nrfxlib:mpsl_changelog>`.
+
+Zephyr
+======
+
+The Zephyr fork in |NCS| (``sdk-zephyr``) contains all commits from the upstream Zephyr repository up to and including ``71ef669ea4a73495b255f27024bcd5d542bf038c``.
+This is the same commit ID as the one used for |NCS| :ref:`v2.1.0 <ncs_release_notes_210>`, :ref:`v2.1.1 <ncs_release_notes_211>`, and :ref:`v2.1.3 <ncs_release_notes_213>`.
```
[the remark this answers is not in the document]
Comment (umapraseeda) [reply doc/nrf/releases/release-notes-2.1.4.rst]: Yes, missed that. Added now.

**Review by peknis — APPROVED** [review]

**Review by wbober — APPROVED** [review]

**Review by nordic-hani — APPROVED** [review]

File: doc/nrf/releases/release-notes-2.1.4.rst [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+=========
+
+This section provides detailed lists of changes by :ref:`library <libraries>`.
+
+Binary libraries
+----------------
+
+* :ref:`liblwm2m_carrier_readme` library:
+
+  * Added workaround for the known issue TNSW-46156.
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-2.1.4.rst:101]: Is this a known issue tracked on our known issues page? If not, can we either add it to the list on the known issues page or add a link here that points to its description? Or just explain where this known issue is coming from and what it is about, if it is not coming from us? I can totally see users going to our known issue page to find out what this is about.

**Review by greg-fer — CHANGES_REQUESTED** [review]

File: doc/nrf/releases/release-notes-2.1.4.rst (hunk 2 above)
Comment (umapraseeda) [reply doc/nrf/releases/release-notes-2.1.4.rst:101]: This known issue is only in the main branch. I think, if using hyperlink, we can only link to 'Other issues'? Correct me if I am wrong, we cannot link to the individual issues, right?

Comment (greg-fer) [inline doc/nrf/releases/release-notes-2.1.4.rst:101]: We can link to individual known issues, yes, with a ref tag -- see for example the individual link tag for DRGN-15852.

As the solution to my comment, I'd suggest the following edit to point the user to the list of known issues for the last version that included this known issue - so it is possible to read its description there.

```suggestion
  * Added workaround for the known issue TNSW-46156, last listed as active in the `known issues for nRF Connect SDK v2.1.3`_.
```

Comment (umapraseeda) [reply doc/nrf/releases/release-notes-2.1.4.rst:101]: @nordic-hani, FYI.

Comment (umapraseeda) [reply doc/nrf/releases/release-notes-2.1.4.rst]: Added this.

**Review by b-gent — APPROVED** [review]

File: doc/nrf/releases/release-notes-2.1.4.rst [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+=========
+
+This section provides detailed lists of changes by :ref:`library <libraries>`.
+
+Binary libraries
+----------------
+
+* :ref:`liblwm2m_carrier_readme` library:
+
+  * Added workaround for the known issue TNSW-46156, last listed as active in the `known issues for nRF Connect SDK v2.1.3`_ .
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-2.1.4.rst:101]:
```suggestion
  * Added workaround for the known issue TNSW-46156, last listed as active in the `known issues for nRF Connect SDK v2.1.3`_.
```

**Review by greg-fer — APPROVED** [review]

File: doc/nrf/releases/release-notes-2.1.4.rst (hunk 3 above)
Comment (umapraseeda) [reply doc/nrf/releases/release-notes-2.1.4.rst]: Fixed.

**Review by rlubos — APPROVED** [review]

**Pull request merged**